File: c++/Easy/Part001/Print String Till Character.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string S;
    char C;
    
    // Input for string S
    getline(cin, S);
    
    // Input for character C
    cin >> C;
    
    size_t pos = S.find(C);
    if (pos != string::npos) {
        cout << S.substr(0, pos) << endl;
    } else {
        cout << S << endl; // If C is not found, print the entire string
    }

    return 0;
}

File: Java/Easy/Part001/Print String Till Character.java
import java.util.Scanner;

public class Hello {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Input for string S
        String S = scanner.nextLine();
        
        // Input for character C
        char C = scanner.next().charAt(0);
        
        int pos = S.indexOf(C);
        if (pos != -1) {
            System.out.println(S.substring(0, pos));
        } else {
            System.out.println(S); // If C is not found, print the entire string
        }
        
        scanner.close();
    }
}
